org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/CodeActionFactory.java

```diff
@@ -169,4 +170,22 @@ public static CodeAction createFile(String title, String docURI, String content,
 
 		return codeAction;
 	}
+
+	/**
+	 * Makes a CodeAction to call a command from the available server commands.
+	 *
+	 * @param title         The displayed name of the CodeAction
+	 * @param commandParams The document URI of the document the command is called on
+	 * @param commandId     The id of the given command to add as CodeAction
+	 * @param diagnostic    The diagnostic that this CodeAction will fix
+	 */
+	public static CodeAction createCommand(String title, List<Object> commandParams, String commandId, Diagnostic diagnostic) {

```

Comment:
To be consistent with Command constructor, move `String commandId` before `List<Object> commandParams`